### PR TITLE
fix no-eol-whitespace

### LINKED ISSUE #4218: Fix false negatives for last line without trailing EOL in no-eol-whitespace
> Clearly describe the bug

There is no warning on the last line, when I provide code without trailing EOL

> Which rule, if any, is the bug related to?

`no-eol-whitespace`

> What CSS is needed to reproduce the bug?

any without trailing EOL, e.g.

```css
a {
  width: 00.10em;
}\u0020\u0020
```

No error on line 3 ☝🏼

```css
a {
  width: 00.10em;
}\u0020\u0020\n
```
error on line 3 ☝🏼

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "no-eol-whitespace": [true, { "ignore": ["empty-lines"] }]
  }
}
```

> Which version of stylelint are you using?

latest on https://stylelint.io/demo

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

REPL

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

no

> What did you expect to happen?

warning without trailing EOL too

> What actually happened (e.g. what warnings or errors did you get)?

no warning

May be related to #3842 
Right now code runs` styleSearch` for EOL, so last line is skipped

### PR DESCRIPTION
**Pull request by vankop**
> Which issue, if any, is this issue related to?

fix #4218 #3842 

> Is there anything in the PR that needs further explanation?

nothing

### CODE REVIEW

File: lib/rules/no-eol-whitespace/index.js [hunk 1]
```diff
@@ -179,6 +211,14 @@ const rule = function(on, options, context) {
         },
         isRootFirst
       );
+
+      const lastEOL = root.raws.after.lastIndexOf("\n");
```
Comment (evilebottnawi) [inline lib/rules/no-eol-whitespace/index.js]: What about `\r`? It is rare use case, but some old os use `\r` as end of line

Comment (vankop) [reply lib/rules/no-eol-whitespace/index.js]: done

File: lib/rules/no-eol-whitespace/__tests__/index.js [hunk 2]
```diff
@@ -383,6 +389,150 @@ testRule(rule, {
...
+      fixed: "// comment 3\ra {color: red}",
+      warnings: [
+        {
+          message: messages.rejected,
+          line: 1,
+          column: 14
+        },
+        {
+          message: messages.rejected,
+          line: 1,
```
Comment (vankop) [inline lib/rules/no-eol-whitespace/__tests__/index.js]: @evilebottnawi looks like we don't support CR =(

**Comment by vankop** [comment]
tests works locally on OSX, will try to restart one more time

**Comment by hudochenkov** [comment]
@vankop do you have rights both in Travis and AppVeyor to restart? If not, tell me where do you not have an access. Closing and re-opening produces unnecessary email spam.

**Comment by vankop** [comment]
@hudochenkov did not find where restart button in AppVeyor, in Travis I have rights

**Comment by hudochenkov** [comment]
When you go to build https://ci.appveyor.com/project/stylelint/stylelint/builds/26918044. You have to be signed with Github and have stylelint selected. You should see “Re-build PR” button.

Screenshot 2019-08-30 at 12 41 14

**Comment by hudochenkov** [comment]
> -   Fixed: `no-eol-whitespace` autofix for within comments ([#4224](https://github.com/stylelint/stylelint/pull/4224)).
> -   Fixed: `no-eol-whitespace` false negatives for last line without trailing EOL ([#4224](https://github.com/stylelint/stylelint/pull/4224)).

**Comment by vankop** [comment]
@hudochenkov I don't have access =(
Снимок экрана 2019-08-30 в 14 26 21

**Comment by hudochenkov** [comment]
@vankop I made some changes in AppVeyor. Could you check again, please?

**Comment by vankop** [comment]
> I made some changes in AppVeyor. Could you check again, please?

Thanks, now it works

**Pull request merged**